apps/web/vibes/soul/form/file-upload/file-upload-item.tsx

```diff
@@ -0,0 +1,209 @@
+'use client';
+
+import { clsx } from 'clsx';
+import {
+  FileArchive,
+  FileAudio,
+  FileCode,
+  FileCog,
+  FileIcon,
+  FileImage,
+  FileText,
+  FileVideo,
+  XIcon,
+} from 'lucide-react';
+import { useId } from 'react';
+
+import { FileUploadProps } from '@/vibes/soul/form/file-upload';
+import { useFileUpload } from '@/vibes/soul/form/file-upload/file-upload-provider';
+import { Button } from '@/vibes/soul/primitives/button';
+
+export interface FileUploadItemProps
+  extends Pick<FileUploadProps, 'uploadingLabel' | 'successLabel' | 'idleLabel' | 'errorLabel'> {
+  file: File;
+}
+
+export function FileUploadItem({
+  file,
+  uploadingLabel = 'Uploading',
+  successLabel = 'Upload complete',
+  idleLabel = 'Ready to upload',
+  errorLabel = 'Error',
+}: FileUploadItemProps) {
+  const id = useId();
+  const statusId = `${id}-status`;
+  const nameId = `${id}-name`;
+  const sizeId = `${id}-size`;
+  const messageId = `${id}-message`;
+
+  const { state, onRemoveFile } = useFileUpload();
+
+  const fileState = state.files.get(file);
+  const fileCount = state.files.size;
+  const fileIndex = Array.from(state.files.keys()).indexOf(file) + 1;
+
+  let statusText: string;
+
+  if (Boolean(fileState?.error)) {
```

Comment:
Instead of coercing, can we do `fileState?.error != null`?

Reply:
curious here, what's the problem with the coercion? I find the single = more troublesome... 🤔